### PR TITLE
Improve Workers docs for Octopus Cloud

### PR DESCRIPTION
**Pull request by samjetski**
[sc-62106]

Aside from the Dynamic Workers page, the current Workers section is exclusively oriented towards **self-hosted Octopus**.

We have some opportunities here:

- Sell the Octopus Cloud offering, to support our Cloud Revenue OKR
  - The DW page doesn't currently link to Octopus Cloud anywhere
  - No other part of Workers mentions DW or cloud, despite it being a significant part of our offering
- Fix the broader [Workers](https://octopus.com/docs/infrastructure/workers) and [Worker Pools](https://octopus.com/docs/infrastructure/workers/worker-pools) docs to include Dynamic Workers as an option
  - For a potential customer, it just looks way more complex than it needs to be
  - For a cloud customer getting started, it leads them away from the easy route. The "Built-In Worker" explanation is misleading, deploying your own tentacle worker is usually unnecessary, and they would have to stumble on Dynamic Workers by accident.
- Clarify what the default pool does (there must always be one, but what is it?), and how it differs across self-hosted and cloud.
- Improve our guidance about selecting worker image versions. We should recommend using `worker-tools` images as first preference over selecting a worker OS version.

Changes required before merging:

- [x] Merge #2024 
- [x] Reflect current default DW pool. Either:
    - [ ] Include cutover date in doc, i.e. "Cloud customers who signed up before... will have Windows as the default pool on new spaces, and on existing spaces unless they have changed it", or
    - [x] Revert default pool info to Windows, and update later when [publishing cut-over date](https://app.shortcut.com/octopusdeploy/story/61174/add-cutover-date-to-public-documentation)
- [x] Update modified date on all pages

This was raised as part of [Make Linux the default Dynamic Worker](https://app.shortcut.com/octopusdeploy/story/37866/make-linux-the-default-dynamic-worker).

### CODE REVIEW

File: src/pages/docs/infrastructure/workers/index.mdx [hunk 1]
```diff
@@ -38,13 +38,17 @@ The following steps always run inside the Octopus Server process (and do not run
...
 A worker receives instruction from the Octopus Server to execute a step, it executes the step using Calamari and returns the logs and any collected artifacts to the Octopus Server.
 
+:::div{.hint}
+Workers are assigned at the start of a deployment or runbook, not at the time the individual step executes.
+:::
+
 There are two kinds of workers you can use in Octopus:
 
-1. [The built-in worker](#built-in-worker) (default)
+1. [The built-in worker](#built-in-worker) (default for self-hosted Octopus)
```
Comment (samjetski) [inline src/pages/docs/infrastructure/workers/index.mdx:47]: Change to "only available"

File: src/pages/docs/infrastructure/workers/index.mdx [hunk 2]
```diff
@@ -38,13 +38,17 @@ The following steps always run inside the Octopus Server process (and do not run
...
 1. [External workers](#external-workers)
```
Comment (samjetski) [inline src/pages/docs/infrastructure/workers/index.mdx:49]: Add 3rd item for "Dynamic Workers" (only available in Octopus Cloud). Also, table?

File: src/pages/docs/infrastructure/workers/index.mdx [hunk 3]
```diff
@@ -38,13 +38,17 @@ The following steps always run inside the Octopus Server process (and do not run
...
 
-:::div{.hint}
-Workers are assigned at the start of a deployment or runbook, not at the time the individual step executes.
+:::div{.success}
+[Octopus Cloud](/docs/octopus-cloud) uses [Dynamic Workers](/docs/infrastructure/workers/dynamic-worker-pools) by default, which provides External Workers running on Ubuntu and Windows VMs. Dynamic Workers are managed by Octopus Cloud, and are included with your Octopus Cloud subscription.
```
Comment (samjetski) [inline src/pages/docs/infrastructure/workers/index.mdx:51]: "External Workers" -> "an on-demand worker"

File: src/pages/docs/infrastructure/workers/index.mdx [hunk 4]
```diff
@@ -67,18 +71,28 @@ Adding a worker to the default worker pool will disable the built-in worker, and
 
 Learn about the security implications and how to configure the [built-in worker](/docs/infrastructure/workers/built-in-worker).
 
+:::div{.warning}
```
Comment (samjetski) [inline src/pages/docs/infrastructure/workers/index.mdx:74]: `.hint`

**Comment by steve-fenton-octopus** [comment]
Hi @samjetski 

I've pushed a little fix for spelling related things, which involved a merge from main (to get the latest dictionary).

**Review by steve-fenton-octopus — APPROVED** [review]
LGTM

**Pull request merged**